### PR TITLE
Hides the Promote Post header in the WC Mobile App

### PR DESCRIPTION
**Pull request by sbarbosa**
We want to hide the Promote post header when the page is loaded from the WooCommerce app. We are already doing it for the jetpack mobile app.

Screenshot 2023-12-12 at 15 54 47

## Proposed Changes

* Add another flag in the server to validate the source against WooCommerce user agent
* When the document is loaded, add the class `is-mobile-app-view` to the body if its Jetpack or WooCommerce mobile app

## Testing Instructions

* Open the live preview and navigate to Tools->Advertising
* In your browser, change the page to a mobile device, and change the user agent to be this one:
```
Mozilla/5.0 (iPhone; CPU iPhone OS 14_6 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Mobile/15E148 wc-android
```
* Refresh the page and verify that the Woo logo is shown on the page load
Screenshot 2023-12-15 at 16 27 11

* Verify that the Advertising dashboard is displayed correctly for a mobile app (no header and menu)
* Click on the Promote post for any of your post and verify that the Widget header is hidden and that the Get started step was not shown

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/server/lib/is-mobile-app/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export function isWpMobileApp( source ) {
+	return /wp-(android|iphone)/.test( source );
+}
+
+/**
+ * Returns whether user is using a WooCommerce mobile app.
+ * @param source string
+ * @returns {boolean} Whether source matches the ones used on the WooCommerce mobile apps.
+ */
+export function isWcMobileApp( source ) {
```
Comment (enejb) [inline client/server/lib/is-mobile-app/index.js:15]: Can we update this to `isWooMobileApp` I think this will would make it more clear.

Comment (sbarbosa) [reply client/server/lib/is-mobile-app/index.js:15]: I followed the naming we already have in Calypso for the WooCommerce app check. 
You can see it in this package: https://github.com/Automattic/wp-calypso/tree/trunk/client/lib/mobile-app

I feel that if we change it here, we will need to change it in the client too, and multiple files are already using it; The refactor will take a good number of files. 

Let me know what you think and if you want to apply that refactor 💪

Comment (enejb) [reply client/server/lib/is-mobile-app/index.js:15]: Thanks for the clarification. @sbarbosa ! 
I thought it was something new we are adding. 

Lets keep things as they are, isWooMobileApp still reads (IMHO) better but we can do that change down the line at a later time if at all. :)

**Review by timur987 — APPROVED** [review]
Thanks, @sbarbosa!

The changes LGTM (assuming you'll apply suggestion from @enejb soon) and works as expected.

The only difference I found was in your claim about "Get started step was not shown", for me it was shown as usual (don't know whether it's inappropriate or not though).

https://github.com/Automattic/wp-calypso/assets/115007291/7f669ae2-3dce-434f-a837-b5c054cd7989

**Review by therocket-gr — APPROVED** [review]
tested the live preview link and it works as described
code changes LGTM

**Comment by therocket-gr** [comment]
> Thanks, @sbarbosa!
> 
> The changes LGTM (assuming you'll apply suggestion from @enejb soon) and works as expected.
> 
> The only difference I found was in your claim about "Get started step was not shown", for me it was shown as usual (don't know whether it's inappropriate or not though).
> 
>  Screen.Recording.2023-12-18.at.14.02.04.mov

for me it doesnt show? i checked my localstorage and i cant find the flag `hideGetStartedMessage`

**Comment by sbarbosa** [comment]
Thanks for the review! 

> The only difference I found was in your claim about "Get started step was not shown", for me it was shown as usual (don't know whether it's inappropriate or not though).

One question here @timur987, how are you changing the user agent? I checked, and your video seems to be using the responsive view of Chrome. I only know how to change the user agent in Chrome by creating a new device, and updating the agent in there. Maybe you are using an extension to change it, or in some other way that only works on page load (it's just a guess). 

I tested this in Chrome and Firefox, and if I use the device mode, it works. Let me know how you are making the test, so I can try to replicate it and maybe debug why at the moment of opening the widget, the User-agent test fails. I see that the page is loaded almost the same as me (for some reason, my page doesn't have the page margin at the top.

**Comment by timur987** [comment]
> > The only difference I found was in your claim about "Get started step was not shown", for me it was shown as usual (don't know whether it's inappropriate or not though).
 
> One question here @timur987, how are you changing the user agent? I checked, and your video seems to be using the responsive view of Chrome. I only know how to change the user agent in Chrome by creating a new device, and updating the agent in there. Maybe you are using an extension to change it, or in some other way that only works on page load (it's just a guess).

Hey @sbarbosa, again, I checked and everything works well except your claim about "Get started step was not shown", in my case it is shown (don't know whether it's a requirement to hide it or not).
 
> I tested this in Chrome and Firefox, and if I use the device mode, it works. Let me know how you are making the test, so I can try to replicate it and maybe debug why at the moment of opening the widget, the User-agent test fails. I see that the page is loaded almost the same as me (for some reason, my page doesn't have the page margin at the top.

I tested it in Chrome. To adjust UA I used Item logo image for [User-Agent Switcher for Chrome](https://chromewebstore.google.com/detail/djflhoibgkdhkhhcedjiklpkjnoahfmg). 

On the videos below first time I tried without device toggling:

https://github.com/Automattic/wp-calypso/assets/115007291/66e8358c-6259-4d1d-9a9e-681a051f6344

The second time I tried with device toggling instead:

https://github.com/Automattic/wp-calypso/assets/115007291/9973822f-f193-4ced-a0e3-33125d55ef3b

In both cases I don't have the top header (as on the image below):
image

I assumed that missing of "Get started" step is a requirement, that's why I highlighted his presence in my testing. Otherwise I do not see any blockers to proceed with this PR.

**Comment by therocket-gr** [comment]
i didnt install any 3rd party extensions.. in chrome there is a way to manually add UA

in developers tools click the 3 dots icon at the far right of your navigation bar and select more tools->network conditions

Screenshot 2023-12-19 at 13 06 19

in the network conditions tab there is a User Agent section (disable the use browser default and enter a custom one below) 

Screenshot 2023-12-19 at 13 06 34

**Comment by sbarbosa** [comment]
Thanks for sharing the testing steps @timur987! 

I found out the discrepancy here. What is happening is that the `User-Agent Switcher for Chrome` extension is only changing the user agent in all of the HTTP requests but not in the browser's navigator object. The server is the one that is adding the `is-mobile-app-view`, which is why you are getting the correct page on load, but then, all of the other checks we do use the `navigator.userAgent` are failing. 

The second test you shared didn't work because you were using the Responsive device in the Chrome inspector.
You can follow this gif to create a custom device using that User-agent. It will only work to simulate mobile devices thought.


The mobile team hasn't mentioned this edge case happening, so the Webview must be taking the correct user agent and generating the proper navigator object. I don't think this will happen in production 👍.

**Comment by timur987** [comment]
Thanks for sharing how to use the tooling within Chrome. That's awesome! TIL.

**Comment by mpkelly** [comment]
Screenshot 2567-01-11 at 14 52 36

Hi @sbarbosa, I think the logo CSS should be scoped as it's all black in on this page.

**Comment by sbarbosa** [comment]
Hello @mpkelly! The logo you mentioned is a different logo than the one that was added to this PR. 
From a quick search, that logo is found [here](https://github.com/Automattic/wp-calypso/blob/dede124e72a96c421ee7a408729f355da9aec70f/client/my-sites/woocommerce/landing-page.tsx#L212).

I am unfamiliar with WooCommerce; maybe you can ping someone from the WooCommerce team who worked on the Landing page in Calypso?

**Comment by mpkelly** [comment]
> Hello @mpkelly! The logo you mentioned is a different logo than the one that was added to this PR. 
> From a quick search, that logo is found [here](https://github.com/Automattic/wp-calypso/blob/dede124e72a96c421ee7a408729f355da9aec70f/client/my-sites/woocommerce/landing-page.tsx#L212).
> 
> I am unfamiliar with WooCommerce; maybe you can ping someone from the WooCommerce team who worked on the Landing page in Calypso?

Hi @sbarbosa, I think the issue is the class name you're using  - it's matching elsewhere. I'm not at my computer right now so can't double-check that. I can take a look again tomorrow.

**Comment by sbarbosa** [comment]
That is strange; styles should be self-contained in the Component. They shouldn't spread to the rest of the app.
Let me check that, just to be sure 👍.

**Comment by sbarbosa** [comment]
Nice catch @mpkelly! You were on the spot! 

I will quickly change this and push a PR with the fix.
Thanks for letting me know. I wasn't aware that the styles were going to spread. 

I will be more careful about this in the future 👍

**Comment by mpkelly** [comment]
>They shouldn't spread to the rest of the app.

Before A8c I worked with CSS-in-JS and forgot about this type of problem. But in Calypso, CSS gets loaded but not unloaded, so we have to be extra careful. I've been caught out by this multiple times already, and it's a common source of bugs. Thanks for fixing it.

**Pull request merged**